src/amber/router/pipe/method.cr

```diff
@@ -0,0 +1,21 @@
+module Amber::Pipe
```

Comment:
Although this is 10 times cleaner, I'm worried that you have a chicken/egg issue.  The pipeline doesn't happen until after the route is determined, but the route is dependent on the method. 

I'm afraid this isn't going to work unless you add the handler before the router.